Attempting to build for ARM does not work

I'm trying to build krist-miner for an arm machine, but running `GOOS=linux GOARCH=arm go get && go build` returned 5 errors:
`./miner.go:152:7: undefined: mineSHA`
`./miner.go:154:7: undefined: mineAVX2`  
`./miner.go:156:7: undefined: mineAVX`
`./miner.go:158:7: undefined: mineSSSE3`
`./miner.go:160:7: undefined: mineARM`

Trying to build for my computer's architecture (amd64) returns no errors and the program runs as expected.

--- a/cpu-miner/cpuid/cpuid_386.s
+++ b/cpu-miner/cpuid/cpuid_386.s
@@ -1,3 +1,5 @@
+// +build 386,!gccgo
+
 // The MIT License (MIT)
 //
 // Copyright (c) 2015 Klaus Post
@@ -19,8 +21,6 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
-
-// +build 386,!gccgo
 
 // func cpuid(op uint32) (eax, ebx, ecx, edx uint32)
 TEXT ·cpuid(SB), 7, $0

--- a/cpu-miner/cpuid/cpuid_amd64.s
+++ b/cpu-miner/cpuid/cpuid_amd64.s
@@ -1,3 +1,5 @@
+// +build amd64,!gccgo
+
 // The MIT License (MIT)
 //
 // Copyright (c) 2015 Klaus Post
@@ -19,8 +21,6 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
-
-// +build amd64,!gccgo
 
 // func cpuid(op uint32) (eax, ebx, ecx, edx uint32)
 TEXT ·cpuid(SB), 7, $0

--- a/cpu-miner/cpuid/cpuid_arm64.go
+++ b/cpu-miner/cpuid/cpuid_arm64.go
@@ -1,5 +1,3 @@
-// +build arm64,linux
-
 // Minio Cloud Storage, (C) 2016 Minio, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -17,10 +15,7 @@
 
 package cpuid
 
-import (
-	"bytes"
-	"io/ioutil"
-)
+import "os"
 
 func cpuid(op uint32) (eax, ebx, ecx, edx uint32) {
 	return 0, 0, 0, 0
@@ -34,16 +29,6 @@ func xgetbv(index uint32) (eax, edx uint32) {
 	return 0, 0
 }
 
-// File to check for cpu capabilities.
-const procCPUInfo = "/proc/cpuinfo"
-
-// Feature to check for.
-const sha256Feature = "sha2"
-
 func haveArmSha() bool {
-	cpuInfo, err := ioutil.ReadFile(procCPUInfo)
-	if err != nil {
-		return false
-	}
-	return bytes.Contains(cpuInfo, []byte(sha256Feature))
+	return os.Getenv("ARMSHA") != ""
 }

--- a/cpu-miner/cpuid/cpuid_other.go
+++ b/cpu-miner/cpuid/cpuid_other.go
@@ -1,3 +1,5 @@
+// +build !386,!amd64,!arm64
+
 // Minio Cloud Storage, (C) 2016 Minio, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -12,8 +14,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-
-// +build !386,!amd64,!arm,!arm64 arm64,!linux
 
 package cpuid
 

--- a/cpu-miner/mine_slow.go
+++ b/cpu-miner/mine_slow.go
@@ -1,7 +1,6 @@
 package main
 
 import (
-	"runtime"
 	"time"
 
 	sha256 "github.com/1lann/sha256-simd"
@@ -41,8 +40,6 @@ func mineSlow(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256blockArm_arm64.go
+++ b/cpu-miner/sha256blockArm_arm64.go
@@ -1,18 +1,42 @@
 // +build !noasm
 
+// Minio Cloud Storage, (C) 2016 Minio, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package main
 
 import (
-	"runtime"
 	"time"
 )
 
-//go:noescape
-func blockArm(h []uint32, message []uint8)
+func mineAVX(proc int) {
+	panic("optimization unsupported")
+}
 
-func mineAVX2(proc int)  {}
-func mineAVX(proc int)   {}
-func mineSSSE3(proc int) {}
+func mineAVX2(proc int) {
+	panic("optimization unsupported")
+}
+
+func mineSHA(proc int) {
+	panic("optimization unsupported")
+}
+
+func mineSSSE3(proc int) {
+	panic("optimization unsupported")
+}
+
+func blockArm(h []uint32, message []uint8)
 
 func mineARM(proc int) {
 	instanceID := generateInstanceID()
@@ -61,8 +85,6 @@ func mineARM(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256blockArm_arm64.s
+++ b/cpu-miner/sha256blockArm_arm64.s
@@ -1,4 +1,4 @@
-//+build !noasm,!appengine
+// +build !noasm
 
 // ARM64 version of SHA256
 

--- a/cpu-miner/sha256blockAvx2_amd64.go
+++ b/cpu-miner/sha256blockAvx2_amd64.go
@@ -3,14 +3,15 @@
 package main
 
 import (
-	"runtime"
 	"time"
 )
 
 //go:noescape
 func blockAvx2(h []uint32, message []uint8)
 
-func mineARM(proc int) {}
+func mineARM(proc int) {
+	panic("optimization unsupported")
+}
 
 func mineAVX2(proc int) {
 	instanceID := generateInstanceID()
@@ -59,8 +60,6 @@ func mineAVX2(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256blockAvx_amd64.go
+++ b/cpu-miner/sha256blockAvx_amd64.go
@@ -3,7 +3,6 @@
 package main
 
 import (
-	"runtime"
 	"time"
 )
 
@@ -57,8 +56,6 @@ func mineAVX(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256blockSha_amd64.go
+++ b/cpu-miner/sha256blockSha_amd64.go
@@ -3,7 +3,6 @@
 package main
 
 import (
-	"runtime"
 	"time"
 )
 
@@ -57,8 +56,6 @@ func mineSHA(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256blockSsse_amd64.go
+++ b/cpu-miner/sha256blockSsse_amd64.go
@@ -3,7 +3,6 @@
 package main
 
 import (
-	"runtime"
 	"time"
 )
 
@@ -57,8 +56,6 @@ func mineSSSE3(proc int) {
 		}
 
 		workerSpeeds[proc] = time.Since(start)
-
-		runtime.Gosched()
 
 		if threadBlock != lastBlock {
 			threadBlock = lastBlock

--- a/cpu-miner/sha256block_other.go
+++ b/cpu-miner/sha256block_other.go
@@ -1,3 +1,5 @@
+// +build !386,!amd64,!arm64
+
 // Minio Cloud Storage, (C) 2016 Minio, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -11,22 +13,25 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
 
-package cpuid
+package main
 
-func cpuid(op uint32) (eax, ebx, ecx, edx uint32) {
-	return 0, 0, 0, 0
+func mineAVX(proc int) {
+	panic("optimization unsupported")
 }
 
-func cpuidex(op, op2 uint32) (eax, ebx, ecx, edx uint32) {
-	return 0, 0, 0, 0
+func mineAVX2(proc int) {
+	panic("optimization unsupported")
 }
 
-func xgetbv(index uint32) (eax, edx uint32) {
-	return 0, 0
+func mineSHA(proc int) {
+	panic("optimization unsupported")
 }
 
-func haveArmSha() bool {
-	return false
+func mineSSSE3(proc int) {
+	panic("optimization unsupported")
+}
+
+func mineARM(proc int) {
+	panic("optimization unsupported")
 }

